CONTRIBUTING.md

```diff
@@ -1,14 +1,49 @@
+# Contributing
+
+Welcome to our Github project! We are happy to have you here. We welcome contributions from anyone and everyone. By participating in this project, you agree to abide by our [Code of Conduct](./code_of_conduct.md).
+
+## Getting started
+
+For code-related instructions, please refer to the [README](./README.md) file. You can find issues in the project's issues page and you can submit pull requests to the project's pull requests page. If you are a first-time contributor, you may find value in filtering the issues that are labeled as "good first issue".
+
+## Documentation
+
+We value good documentation. Please ensure that any code you contribute is well-documented. This includes adding comments to your code, adding docstrings to functions and modules, and adding any necessary documentation to the project's README.
+
+Although keep in mind that the code should be straightforward and provide enough context for the reader to understand what it does. One should consider code readability as a priority over documentation. And documentation should be present to fill in the gaps, as needed.
+
+
+Another good way of documenting the behaviour of a feature should include tests. Unit tests are a great way to include documentation in the codebase. If you are adding a new feature, please consider adding tests for it. We appreciate good test coverage.
+
+For more information on testing, please refer to the [Testing](./testing.md) section.
+
 ## Filing an issue
-When filing an issue on the EdEHR project, please provide these details:
+When filing an issue on the project, please provide these details:
 
 A comprehensive list of steps to reproduce the issue.
 What you're expecting to happen compared with what's actually happening.
-Screenshots demonstrating the bug.
+Screenshots and video recordings demonstrating the bug are also very helpful.
 
 Please remember to format code using triple backticks (`) so that it is neatly formatted when the issue is posted.
 
 Any issue that is open for 14 days without actionable information or activity will be marked as "stalled" and then closed. Stalled issues can be re-opened if the information requested is provided.
 
+When creating an issue, please use the appropriate template. If you are unsure which template to use, please use the "Bug report" template.
+Other templates that are valuable include: 
+
+- Feature request
+- Documentation Update
+- Features to be added
```

Comment:
We have these templates:
Bug report- Create a report to help us improve
Custom issue template - Fully spec'ed out feature
Small task template- A task to support a larger issue